4_de_analysis/nbconverted/2_de_pilot.py

```diff
@@ -0,0 +1,218 @@
+
+
+# In[30]:
+
+
+identifier = 'NF2_KIRP'
+# adjusted p-value threshold
+alpha = 0.05
+
+none_one_de_results = pd.read_csv(
+    cfg.de_output_dir / 'DE_stats_{}_none_one.txt'.format(identifier), sep='\t'
+)
+print(none_one_de_results.shape)
+none_one_de_results.head()
+
+
+# In[31]:
+
+
+one_both_de_results = pd.read_csv(
+    cfg.de_output_dir / 'DE_stats_{}_one_both.txt'.format(identifier), sep='\t'
+)
+print(one_both_de_results.shape)
+one_both_de_results.head()
+
+
+# In[32]:
+
+
+none_both_de_results = pd.read_csv(
+    cfg.de_output_dir / 'DE_stats_{}_none_both.txt'.format(identifier), sep='\t'
+)
+print(none_both_de_results.shape)
+none_both_de_results.head()
+
+
+# In[33]:
+
+
+none_one_de_count = (
+    (none_one_de_results.padj < alpha).sum()
+)
+one_both_de_count = (
+    (one_both_de_results.padj < alpha).sum()
+)
+none_both_de_count = (
+    (none_both_de_results.padj < alpha).sum()
+)
+
+print('DE genes for {} none vs. one:'.format(identifier), none_one_de_count)
+print('DE genes for {} one vs. both:'.format(identifier), one_both_de_count)
+print('DE genes for {} none vs. both:'.format(identifier), none_both_de_count)
+
+
+# In[34]:
+
+
+import matplotlib.pyplot as plt
+import seaborn as sns
+
+sns.set({'figure.figsize': (10, 8)})
+fig, axarr = plt.subplots(2, 1)
+
+sns.kdeplot(data=none_one_de_results.pvalue, label='none/one', ax=axarr[0])
```

Comment:
I'm a little lost on the terminology here, but you're expecting to see that you have more DEGs when you compare samples with 0 hits to 2-hits (none/both curve) versus when you compare 0-hit and 1-hit samples (none/one curve)?

Reply:
You mention doing some sort of gene-level comparison in your slides, but I'm not sure how that would work -- are you planning to compare p-values across different experiments?

Reply:
> but you're expecting to see that you have more DEGs when you compare samples with 0 hits to 2-hits (none/both curve) versus when you compare 0-hit and 1-hit samples (none/one curve)?

It depends on the gene, that much is pretty well known. The Park et al. paper that I linked in my description classifies genes into "class 1" (one-hit) and "class 2/3/4" (two-hit) cancer drivers.

So for the class 1 genes I would expect 1-hit and 2-hit samples to be very similar (fewer DEGs) and 0-hit and 1-hit samples to be very different (more DEGs), since it only takes a single hit to "activate" the gene as a cancer driver and the second hit would be redundant.

For the class 2/3/4 genes I would expect the opposite - because they're two-hit genes they require two mutations to be "activated" as a cancer driver, so I would expect 1-hit and 2-hit samples to be very different (many DEGs) and 0-hit and 1-hit samples to be similar.

> You mention doing some sort of gene-level comparison in your slides, but I'm not sure how that would work -- are you planning to compare p-values across different experiments?

Yeah, I'm not sure yet either. After talking to Casey I think we're going to abandon the DE approach in favor of something that more explicitly corrects for sample size, since DEG count is so dependent on the number of samples in each group.

There are lots of ways to define similarity/distance between groups of samples in gene expression space - we liked DE because it's not dependent on the sorts of modeling/parameter tuning decisions that our mutation prediction classifiers are dependent on, but there are probably other methods that are similar and not as sensitive to sample size.